src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -580,12 +580,14 @@ public function formAction($id, Request $request)
             $preview_url_deactive = $adminProductWrapper->getPreviewUrl($product, false);
             $preview_url = $adminProductWrapper->getPreviewUrlDeactivate($preview_url_deactive);
         }
+        
+        $shopIdsList = $shopContext->getContextListShopID();
```

Comment:
Why getting every shops IDs as you have already the good one inside the legacy context ?

Reply:
You said it, "legacy context" in the new part of the code. The objective is not to rely directly on the context properties when possible, you can't certify the data isn't altered.